projects/distributed-tracing/src/shared/dashboard/interaction/detail-sheet/detail-sheet-interaction-handler.service.ts

```diff
@@ -0,0 +1,31 @@
+import { PortalInjector } from '@angular/cdk/portal';
+import { Injectable, Injector } from '@angular/core';
+import { OverlayService, PopoverRef, SheetSize } from '@hypertrace/components';
+import {
+  DetailSheetInteractionContainerComponent,
+  DETAIL_SHEET_INTERACTION_MODEL
+} from './container/detail-sheet-interaction-container.component';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class DetailSheetInteractionHandlerService {
+  public constructor(private readonly injector: Injector, private readonly overlayService: OverlayService) {}
+
+  public showSheet(detailModel: object, sheetSize: SheetSize = SheetSize.Medium): PopoverRef {
+    return this.overlayService.createSheet(
+      {
+        content: DetailSheetInteractionContainerComponent,
+        size: sheetSize
+      },
+      this.buildDetailOverlayInjector(detailModel)
+    );
+  }
+
+  private buildDetailOverlayInjector(detailModel: object): Injector {
+    return new PortalInjector(
```

Comment:
Up to you, but I've actually moved away from using portal injector when not directly work with CDK - newer angular APIs make it easy enough to use injector directly -
```
    return Injector.create({
      providers: [
        {
          provide: DETAIL_SHEET_INTERACTION_MODEL,
          useValue: detailModel
        }
      ],
      parent: this.injector
    });
```

Reply:
Sure. Will use this